libs/vertexai/langchain_google_vertexai/embeddings.py

```diff
@@ -375,10 +387,7 @@ def embed_query(self, text: str) -> List[float]:
         Returns:
             Embedding for the text.
         """
-        if self.model_type != GoogleEmbeddingModelType.TEXT:
-            raise NotImplementedError("Not supported for multimodal models")
-        embeddings = self.embed([text], 1, "RETRIEVAL_QUERY")
-        return embeddings[0]
+        return self.embed([text], 1, "RETRIEVAL_QUERY")[0]
```

Comment:
I think this needs to only add the "RETRIEVAL_QUERY" if the model is text-only and supports those types.

I don't think those embedding types work with multimodal

Reply:
I wanted to keep the logic the same at this point to take advantage of the code that handle batches and docs and that is currently unified. It's just in the moment to prepare the request and call the LLM that the model type is checked and this is ignored if the model is multimodal. I wanted to maintain the same signature and just allow multimodal models to be used.